fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/foundationdb/FDBRecordStoreTest.java

```diff
@@ -1177,6 +1177,29 @@ public void testNoUnion() {
         }
     }
 
+    @Test
+    public void testDryRunSaveRecord() throws Exception {
+        CompletableFuture<FDBStoredRecord<Message>> future1;
+        TestRecords1Proto.MySimpleRecord.Builder recBuilder = TestRecords1Proto.MySimpleRecord.newBuilder();
+        TestRecords1Proto.MySimpleRecord record1 = recBuilder.setRecNo(1).build();
+        TestRecords1Proto.MySimpleRecord record2 = recBuilder.setRecNo(2).setNumValue3Indexed(3).build();
+        try (FDBRecordContext context = openContext()) {
+            openSimpleRecordStore(context);
+            future1 = recordStore.dryRunSaveRecordAsync(record1, FDBRecordStoreBase.RecordExistenceCheck.ERROR_IF_EXISTS);
+            // assert returns the message to be saved
+            assertEquals(record1, future1.get().getRecord());
+            assertEquals(record2, recordStore.saveRecord(record2).getRecord());
+            commit(context);
+        }
+        // assert dryRun doesn't save the record
+        try (FDBRecordContext context = openContext()) {
+            openSimpleRecordStore(context, metaDataBuilder -> metaDataBuilder.updateRecords(TestRecords1Proto.getDescriptor()));
+            assertNull(recordStore.loadRecord(Tuple.from(1)));
+            assertEquals(record2, recordStore.loadRecord(Tuple.from(2)).getRecord());
+            commit(context);
+        }
+    }
```

Comment:
We should probably have tests that validate that the update (and insert and potentially delete) plans honor the dry-run flag in the `ExecuteProperties`. There are some tests of those plans in `FDBModificationQueryTest`, and it would be nice to execute either those same tests or similar new tests in dry-run mode and validate that database mutations are not made.